src/compiler/checker/lines_8000_12000.rs

```diff
@@ -88,17 +88,27 @@ impl TypeChecker {
         if let Some(declared_type) = declared_type {
             return Some(self.add_optionality(declared_type, Some(is_property), Some(is_optional)));
         }
-        unimplemented!()
+
+        if has_only_expression_initializer(declaration)
+            && declaration
+                .as_has_expression_initializer()
+                .maybe_initializer()
+                .is_some()
+        {
+            let type_ = self.check_declaration_initializer(declaration, None);
+            let type_ = self.widen_type_inferred_from_initializer(declaration, type_);
+            return Some(self.add_optionality(type_, Some(is_property), Some(is_optional)));
+        }
+
+        None
     }
 
     pub(super) fn get_widened_type_for_variable_like_declaration(
         &self,
         declaration: &Node,
     ) -> Rc<Type> {
-        self.widen_type_for_variable_like_declaration(
-            self.get_type_for_variable_like_declaration(declaration),
-            declaration,
-        )
+        let type_ = self.get_type_for_variable_like_declaration(declaration);
+        self.widen_type_for_variable_like_declaration(type_, declaration)
```

Comment:
I think this change ended up being unnecessary, at one point I was trying to change some `TypeChecker` methods to take `&mut self` but unsurprisingly that "bubbled up" to somewhere where that wasn't going to work (eg where a closure that captures a mutable `self` reference gets passed to another `TypeChecker` method) so backed it out (I don't think I had to then eg wrap any `TypeChecker` fields in a `RefCell` so maybe there were just "dangling" existing `&mut self`'s that didn't need to be `&mut self`'s)